core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -537,10 +537,19 @@ private[spark] class ExecutorAllocationManager(
       }
     }
 
+    override def onTaskResubmit(taskResubmit: SparkListenerTaskResubmit): Unit = {
+      val stageId = taskResubmit.stageId
+      allocationManager.synchronized {
+        val num = stageIdToNumTasks.getOrElse(stageId, 0)
+        stageIdToNumTasks.update(stageId, num + 1)
+      }
+    }
+
```

Comment:
I don't see why this is necessary -- when an executor goes down, and stages get resubmitted, [the stage gets resubmitted](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L1088) which will [post another `StageSubmitted` event](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L874).  It looks to me like that will already handle this -- so won't this double count?